engine/include/Engine/Core/Rendering/Window/IWindow.hpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2021 Amara Sami, Dallard Thomas, Nardone William, Six Jonathan
+ * This file is subject to the LGNU license terms in the LICENSE file
+ *	found in the top-level directory of this distribution.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace Engine::Core::Renderering
+{
+struct WindowCreateArg
+{
+    std::string name;
+    size_t height;
+    size_t width;
+};
```

Comment:
1. Using a struct for initialization does not feel necessary here. Passing a `struct` is a clean way to avoid a multi-line constructor, with hard-to-guess parameters. But here, something like that would already be pretty clear:
```C++
GLFWWindow win{"My window", 1200u, 780u};
// Or just
initialize("My window", 1200u, 780u);
```

2. `size_t` is rather used to mesure data structure sizes, for pointers, or for an index in a collection. Under 64-bit machines, on most compilers (at least msvc, clang and gcc), a `size_t` is 8 bytes, which means it can range from 0 to 18.446.744.073.709.551.616‬. So I think an unsigned short, which ranges from 0 to 65.536, would be enough

Reply:
1: This technical allows us to expand easily the struct with the window parameters. The windows can easily take more argument. By default, I just set the name and its size but in the future, we may want V-sync, docking, or size control parameter.  It's easier to add it with default value in struct without chang all reference of this function.

2: I will chang it for an unsigned short